### ISSUE TITLE
Binary Tree

### ISSUE THREAD

**Issue by geekquad** [geekquad/AlgoBook#300]
Implementation of the following in Cpp:

- [ ] AVL Tree

- [ ] Basic Binary Tree

- [ ] Binary Search Tree

- [ ] Binary Tree traversal

- [ ] Recursive Binary Search Tree

- [ ] Lazy Segment Tree

- [ ] Fenwick Tree

- [ ] Non Recursive Segment Tree

- [ ] Red-Black Tree

**Comment by Iampradeepkundra** [comment]
Assign me binary search tree and binary tree traversal

**Comment by Its-Puja-Singh** [comment]
Assign me the AVL Tree, Basic Binary Tree and, Recursive Binary Search Tree

**Comment by geekquad** [comment]
Assigning Binary Search Tree and Binary Tree Traversal to @Iampradeepkundra.
Assigning  AVL Tree, Basic Binary Tree and, Recursive Binary Search Tree to @Its-Puja-Singh

**Comment by aniketprasadgit** [comment]
I want to work here !

**Comment by geekquad** [comment]
Assigning Lazy Segment Tree, Fenwick Tree, and Non-Recursive Segment Tree to @aniketprasadgit.

**Comment by aniketprasadgit** [comment]
I want to work here on  **Red-Black Tree** please assign to me if possible.

**Comment by preyam2002** [comment]
I would like to work on fenwick tree

**Comment by Aarushi11H** [comment]
hey!
I would like to work on Recursive Binary Search Tree

**Comment by Abhi-Kmr2046** [comment]
I would like to work on all of them. Please assign me whichever is not taken.

**Comment by geekquad** [comment]
Assigning Fenwick Tree to @preyam2002.
Assigning Recursive Binary Search Tree to @Aarushi11H. 
Assigning AVL tree to @Abhi-Kmr2046. 
Assigning Non-Recursive Segment Tree, Red-Black Tree, and Lazy Segment Tree to @Abhi-Kmr2046.

**Comment by geekquad** [comment]
Assigning Fenwick Tree to @preyam2002.
Assigning Recursive Binary Search Tree to @Aarushi11H.
Assigning AVL tree to @BRJbibhu10. 
Assigning Non-Recursive Segment Tree, Red-Black Tree, and Lazy Segment Tree to @Abhi-Kmr2046.

**Comment by Iampradeepkundra** [comment]
i was assigned for BST and BTT and I Implemented for the same as well you can see the source files in directory i think accidentely uh unassiged me!

**Comment by geekquad** [comment]
Yes, @Iampradeepkundra. Those were assigned and completed by you. As such you were unassigned.

**Comment by Pranjal-dew** [comment]
hi there, can I contribute to this issue?
